From 9f5a1223b2fb234b866a531690698988ccf188f9 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Mon, 24 Feb 2014 14:31:46 -0600
Subject: [PATCH 284/285] input: touchscreen: pixcir: also trigger IRQ on
 rising edge

We want to make sure we never miss penup/down events
and the easiest way is to also trigger IRQ on the rising
edge.

This will make sure that when we touch or release the
touchscreen surface, an IRQ will be fired and our ISR
handler will be called, thus avoiding missed events.

Fixes #D-01570 - Cap touch screen issue

Tested-by: Franklin Cooper Jr <fcooper@ti.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/input/touchscreen/pixcir_i2c_ts.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/input/touchscreen/pixcir_i2c_ts.c b/drivers/input/touchscreen/pixcir_i2c_ts.c
index a7725b9..f02ad52 100644
--- a/drivers/input/touchscreen/pixcir_i2c_ts.c
+++ b/drivers/input/touchscreen/pixcir_i2c_ts.c
@@ -480,7 +480,8 @@ static int pixcir_i2c_ts_probe(struct i2c_client *client,
 	}
 
 	error = devm_request_threaded_irq(dev, client->irq, NULL, pixcir_ts_isr,
-				     IRQF_TRIGGER_FALLING | IRQF_ONESHOT,
+				     IRQF_TRIGGER_RISING | IRQF_TRIGGER_FALLING
+				     | IRQF_ONESHOT,
 				     client->name, tsdata);
 	if (error) {
 		dev_err(dev, "failed to request irq %d\n", client->irq);
-- 
1.7.5.4

